Content.Server/Bed/Sleep/SleepingSystem.cs

```diff
@@ -175,7 +178,8 @@ private void OnInit(EntityUid uid, ForcedSleepingComponent component, ComponentI
         /// </summary>
         public bool TrySleeping(EntityUid uid)
         {
-            if (!HasComp<MobStateComponent>(uid))
+            //Long term, there should probably be a more universal component/attribute that prevents unconventional mob types from sleeping so they don't need to be listed here individually
+            if (!HasComp<MobStateComponent>(uid) || _mobStateSystem.IsDead(uid) || HasComp<ZombieComponent>(uid) )
```

Comment:
Short term, this should be an event that mobstatesystem and zombiesystem handle, instead of directly checking for them here.

Reply:
I'll remove the zombie part and do that separately, since the same deal comes up for the sneezes too

Reply:
Issue not resolved, this should fire an event to check.

Reply:
Would it be okay if we just removed the UncontrollableSnough and Narcolepsy components when converting a mob to a zombie? This is only an issue because zombies can inherit these components from their crewmember "base". And they can't turn back, so they will never need them again

Reply:
> Would it be okay if we just removed the UncontrollableSnough and Narcolepsy components when converting a mob to a zombie? This is only an issue because zombies can inherit these components from their crewmember "base". And they can't turn back, so they will never need them again

No, seriously, just make an event.

Reply:
I have made the checks event-based